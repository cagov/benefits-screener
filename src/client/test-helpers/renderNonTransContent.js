import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
