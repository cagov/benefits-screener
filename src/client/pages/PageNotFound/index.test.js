import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";

describe("<PageNotFound />", () => {
  it("renders a 404 page", async () => {
    const wrapper = renderNonTransContent(Component);

    expect(wrapper).toMatchSnapshot();
  });
});
