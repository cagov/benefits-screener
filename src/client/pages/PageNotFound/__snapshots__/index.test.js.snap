// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNotFound /> renders a 404 page 1`] = `
<div
  id="overflow-wrapper"
>
  <Header />
  <main>
    <div
      className="container p-4"
    >
      <h1>
        Page Not Found
      </h1>
      <p>
        The page or form you requested cannot be found. If your link worked before but does not work now, then this page has moved or was deleted as part of our ongoing efforts to make our website work better for you.
      </p>
    </div>
  </main>
  <Footer />
</div>
`;
